File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    let result_arr = []
    numbers.forEach((number) => {
        let result_arr_temp = []
        if(result_arr.length == 0) {
            result_arr_temp = [number, -number]
        } else {
            result_arr.forEach((result_item) => {
                result_arr_temp.push(result_item + number)
                result_arr_temp.push(result_item - number)
            })
        }
        result_arr = result_arr_temp
    })
    result_arr.forEach((number) => {
        if(number == target) {
            answer = answer + 1
        }
    })
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    private int[] numbers;
    private int target;
    private int answer = 0;
    
    public int solution(int[] numbers, int target) {
        this.numbers = numbers;
        this.target = target;
        // 현재까지 더한 결과, 다음 정해야 할 숫자
        dfs(0, 0);
        
        return answer;
    }
    
    private void dfs(int sum, int idx) {
        if (idx == numbers.length) {
            if(sum == target) {
                answer++;    
            }
            return;
        }
        
        dfs(sum + numbers[idx], idx + 1);
        dfs(sum - numbers[idx], idx + 1);
    }
}